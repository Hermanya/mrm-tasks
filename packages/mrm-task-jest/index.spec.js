/* eslint-disable no-console */
'use strict';

jest.mock('fs');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));
jest.mock('mrm-core/src/npm', () => ({
	install: jest.fn(),
	uninstall: jest.fn(),
}));

const fs = require.requireActual('fs');
const path = require('path');
const getConfigGetter = require('../../mrm').getConfigGetter;
const vol = require('memfs').vol;
const task = require('./index');

const json = o => JSON.stringify(o, null, '  ');

const packageJson = json({
	name: 'unicorn',
});

afterEach(() => vol.reset());

it('should add Jest', () => {
	vol.fromJSON({
		'/package.json': packageJson,
	});

	task(getConfigGetter({}));

	expect(vol.toJSON()).toMatchSnapshot();
});

it('should add a basic test case when index.js file is present', () => {
	vol.fromJSON({
		[`${__dirname}/templates/test.js`]: fs.readFileSync(path.join(__dirname, 'templates/test.js')),
		'/package.json': packageJson,
		'/index.js': '',
	});

	task(getConfigGetter({}));

	expect(vol.toJSON()['/test.js']).toMatchSnapshot();
});
