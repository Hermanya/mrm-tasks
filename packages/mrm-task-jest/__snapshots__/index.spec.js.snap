// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add Jest 1`] = `
Object {
  "/.gitignore": "coverage/",
  "/.npmignore": "__tests__/",
  "/package.json": "{
  \\"name\\": \\"unicorn\\",
  \\"scripts\\": {
    \\"test:jest\\": \\"jest\\",
    \\"test:watch\\": \\"jest --watch\\",
    \\"test:coverage\\": \\"jest --coverage\\",
    \\"test\\": \\"npm run test:jest\\"
  }
}",
}
`;

exports[`should add a basic test case when index.js file is present 1`] = `
"it('should work please', () => {
	const result = 42;
	expect(result).toBeTruthy();
});
"
`;

exports[`should not overwrite test.js 1`] = `"still here"`;
